_1password-gui: 8.0.33-53.BETA -> 8.0.34-stable

savannidgerinel opened the pull request:
This includes an overhaul of the install script, using patchelf to set
the interpreter and the libraries. This reflects a lot of what I've
learned about electron + nixos over the last few months.

This change is motivated by the fact that 1Password's version of
electron changed, probably leading to crashes that I've been seeing when
I try to update the Nix derivation to the latest versions.

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

savannidgerinel committed:
_1password-gui: 8.0.33-52 -> 8.0.34

This includes an overhaul of the install script, using patchelf to set
the interpreter and the libraries. This reflects a lot of what I've
learned about electron + nixos over the last few months.

This change is motivated by the fact that 1Password's version of
electron changed, probably leading to crashes that I've been seeing when
I try to update the Nix derivation to the latest versions.

Co-authored-by: Daniël de Kok <me@github.danieldk.eu>

grahamc reviewed (commented):
Typically the commit messages for updates like this follow the pattern of:

"attribute_name: oldversion -> newversion"

which in this case would be:

> _1password-gui: 8.0.33-53.BETA -> 8.0.34

I don't have many opinions on the rest of the packaging here. I like that you're using `lib.makeLibraryPath` in the expression, outside of the string interpolation.

Also, thanks for adding yourself as a maintainer!

r-rmcgibbo commented:
Result of `nixpkgs-review pr 123515` at 15394ebb run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>_1password-gui</li>
    <li>tests.trivial</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/applications/misc/1password-gui/default.nix:77:3:
```
   |
77 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

savannidgerinel commented:
> Typically the commit messages for updates like this follow the pattern of:
> 
> "attribute_name: oldversion -> newversion"

Fixed!

cole-h commented on pkgs/applications/misc/1password-gui/default.nix:
```diff
@@ -1,61 +1,92 @@
-      --add-flags "$out/share/${pname}/resources/app.asar" \
-      --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath runtimeLibs}"
+  installPhase = ''
+    mkdir -p $out/bin $out/share/1password
```
Please add `runHook preInstall` to the top, and `runHook postInstall` to the bottom of the `installPhase`.

As it is now, if anybody ever overrides the package and tries to do something in `preInstall`, nothing will happen. Once you add the `runHook`s, then this supposed override will work.

savannidgerinel replied:
Since I'm about to have more than one commit, should I squash them and apply that title to the one resulting commit?

savannidgerinel replied:
i didn't at all know about runHook commands! Thanks!

cole-h replied:
Yes, please!

cole-h commented on pkgs/applications/misc/1password-gui/default.nix:
```diff
@@ -64,14 +95,10 @@ stdenv.mkDerivation rec {
     description = "Multi-platform password manager";
     longDescription = ''
```
This seems to be duplicating the `description` now that the "disclaimer" is gone (woo!) -- I think it's fair to remove it :)

cole-h reviewed (commented):
By the way, you only fixed the PR title; not the commit title itself!

I would suggest: `_1password-gui: 8.0.33-53.BETA -> 8.0.34` (drop the `-stable`, since it's not a part of the `version` attribute in the package itself).

cole-h commented:
Sorry to be a broken record, but: looks like the commit title didn't change :grimacing:

savannidgerinel commented:
> Sorry to be a broken record, but: looks like the commit title didn't change grimacing

Maybe this time I actually fixed it. :|

cole-h commented on pkgs/applications/misc/1password-gui/default.nix:
```diff
@@ -1,77 +1,105 @@
+, libXfixes
+, libXrandr
+, libxshmfence
+, lib
```
Yes indeed, much better!

One final nit: `lib` should always be the first input.

The rest LGTM :)

savannidgerinel commented:
I'm hoping for a review from @danieldk or @timstott . And anyone else who has bundled closed source electron apps. I used the Slack app as a bit of a model in my overhaul here, and if this code is good I'm going to apply the same thing to another electron app.

timstott reviewed (approved):
Fantastic work @savannidgerinel. I tested basic functionality on Gnome 3.38 :ok_hand:.
- Can login
- Sync works
- Can create/edit item
- Can copy password
- Can show password in large

Thank you for this

danieldk commented on pkgs/applications/misc/1password-gui/default.nix:
```diff
@@ -1,77 +1,105 @@
 { lib
-, stdenv
```
LGTM. I am making some changes in the ordering of arguments to make them more canonical.

danieldk reviewed (changes requested):
The changes look like a large improvement to me. I made some small changes, the first two are minor nitpicks:

1. Changed the title of the commit message, the first item should ideally be the exact attribute name.
2. Changed the argument order. The canonical order in nixpkgs is: `lib`/`stdenv`, functions used in the derivation, native build inputs, build inputs, check inputs. Within each group, they are typically alphabetically sorted.
3. I moved `rpath` into the `installPhase` scope with a `let`-binding. Having `rpath` as an argument to `stdenv` adds it as an environment variable in the builder. Scoping it keeps the environment clean :).

Though, it seems that the icons and XDG desktop entries are not installed anymore (see the comment). Is there are particular reason for removing them?

savannidgerinel commented on pkgs/applications/misc/1password-gui/default.nix:
```diff
@@ -2,76 +2,107 @@
-    # Desktop file.
-    install -Dt $out/share/applications resources/${pname}.desktop
-    substituteInPlace $out/share/applications/${pname}.desktop \
-      --replace 'Exec=/opt/1Password/${pname}' 'Exec=${pname}'
```
And done. Sorry about that. Running sway/i3 means I don't see those and I forget the importance.

savannidgerinel commented:
> The changes look like a large improvement to me. I made some small changes, the first two are minor nitpicks:
> 
>     1. Changed the title of the commit message, the first item should ideally be the exact attribute name.
> 
>     2. Changed the argument order. The canonical order in nixpkgs is: `lib`/`stdenv`, functions used in the derivation, native build inputs, build inputs, check inputs. Within each group, they are typically alphabetically sorted.
> 
>     3. I moved `rpath` into the `installPhase` scope with a `let`-binding. Having `rpath` as an argument to `stdenv` adds it as an environment variable in the builder. Scoping it keeps the environment clean :).
> 
> 
> Though, it seems that the icons and XDG desktop entries are not installed anymore (see the comment). Is there are particular reason for removing them?

Damn, I just stomped some of your changes. :( Let me see if I can rebuild them again.

savannidgerinel commented:
> Damn, I just stomped some of your changes. :( Let me see if I can rebuild them again.

Okay, I think I restored your work now. Please give it a quick check to make sure it's right.
